File: 백준/Bronze/2581. 소수/소수.cc
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int M, N;
    cin >> M >> N;

    vector<bool> isPrime(N + 1, true); // 0~N까지 소수 여부

    isPrime[0] = isPrime[1] = false; // 0과 1은 소수가 아님

    // 에라토스테네스의 체
    for (int i = 2; i * i <= N; i++) {
        if (isPrime[i]) {
            for (int j = i * i; j <= N; j += i) {
                isPrime[j] = false; // i의 배수는 소수가 아님
            }
        }
    }

    int sum = 0;
    int minPrime = -1;

    // M~N 사이 소수 합과 최소 소수 찾기
    for (int i = M; i <= N; i++) {
        if (isPrime[i]) {
            sum += i;
            if (minPrime == -1) minPrime = i; // 최소 소수 저장
        }
    }

    if (minPrime == -1) { // 소수가 없는 경우
        cout << -1 << "\n";
    } else {
        cout << sum << "\n" << minPrime << "\n";
    }

    return 0;
}

File: 백준/Bronze/2581. 소수/소수.py
M = int(input())
N = int(input())

# 0~N까지 소수 여부 True로 초기화
is_prime = [True] * (N + 1)
is_prime[0] = is_prime[1] = False  # 0과 1은 소수가 아님

# 에라토스테네스의 체
for i in range(2, int(N ** 0.5) + 1):
    if is_prime[i]:
        for j in range(i * i, N + 1, i):
            is_prime[j] = False  # i의 배수는 소수가 아님

# M~N 사이 소수만 추출
primes = [i for i in range(M, N + 1) if is_prime[i]]

# 결과 출력
if primes:
    print(sum(primes))   # 소수 합
    print(primes[0])     # 최소 소수
else:
    print(-1)            # 소수가 없는 경우
